platforms/platforms.go

```diff
@@ -0,0 +1,129 @@
+package platforms
+
+import (
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/containerd/containerd/errdefs"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type platformKey struct {
+	os      string
+	arch    string
+	variant string
+}
+
+var (
+	selectorRe = regexp.MustCompile(`^[A-Za-z0-9_-]$`)
```

Comment:
I think this needs a `*` for allowing multiple characters: `^[A-Za-z0-9_-]*$`

Reply:
... This was completely broken ...